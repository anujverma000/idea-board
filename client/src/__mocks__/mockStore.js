import configureStore from 'redux-mock-store'
import thunk from "redux-thunk";
import rootReducer from "../reducers";

const initialState= {
    idea: {
        list: []
    }
}
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);
store.replaceReducer(rootReducer);

export default store;