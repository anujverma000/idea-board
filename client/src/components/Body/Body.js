import React from "react";
import { connect } from 'react-redux';
import styled from "styled-components";
import IdeaBoard from "../IdeaBoard/IdeaBoard";
import StartScreen from '../StartScreen/StartScreen'

const Body = ({ ideaList }) => {
  return(
      <Wrapper>
        {ideaList.length ?   <IdeaBoard/> : <StartScreen/>}
      </Wrapper>
  )
}


/**
 *  STYLES
 */
const Wrapper = styled.div`
  background-color: ${props => props.theme.backgroundColor};
`;


const mapStateToProps = state => ({
  ideaList: state.idea.list || [],
});

export default connect(mapStateToProps)(Body);