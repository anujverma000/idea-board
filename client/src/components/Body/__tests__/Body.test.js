import React from 'react';
import renderer from 'react-test-renderer';
import Body from '../Body'
import mockStore from "../../../__mocks__/mockStore";
import { Provider } from "react-redux";

it('renders correctly', () => {
  const tree = renderer.create(
      <Provider store={mockStore}><Body /></Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
