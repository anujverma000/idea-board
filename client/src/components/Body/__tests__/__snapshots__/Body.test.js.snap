// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-dxgOiQ fxKPQl"
>
  <div
    className="sc-jzJRlG jTGdR"
  >
    <img
      alt="Welcome"
      className="sc-cSHVUG Jeizk"
      src="welcome.png"
    />
    <h2
      className="sc-kAzzGY iOujjq"
    >
      More Than An Idea Board
    </h2>
    <p
      className="sc-chPdSV iftqCN"
    >
      Clear visible mammer to brainstorm, create plans, manage projects and more.
    </p>
    <button
      className="sc-VigVT hZJfLK"
      onClick={[Function]}
    >
      Add new idea
    </button>
    <div
      className="sc-kpOJdX jePtxa"
    >
      Highlighted Feature List
    </div>
    <ul
      className="sc-kgoBCf ecWeAn"
    >
      <li
        className="sc-kGXeez dOdpmF"
      >
        Instant Add/Edit
      </li>
      <li
        className="sc-kGXeez dOdpmF"
      >
        Update Notifications
      </li>
      <li
        className="sc-kGXeez dOdpmF"
      >
        Permanent storage
      </li>
      <li
        className="sc-kGXeez dOdpmF"
      >
        Offline availablity
      </li>
      <li
        className="sc-kGXeez dOdpmF"
      >
        Multi facet sorting
      </li>
      <li
        className="sc-kGXeez dOdpmF"
      >
        Mobile Friendly
      </li>
    </ul>
  </div>
</div>
`;
